### ISSUE TITLE
Compact Serializer Config NOT Created Correctlly

### ISSUE THREAD

**Issue by dalcorta** [hazelcast/hazelcast-platform-operator#1023]
Hazelcast Operator Version: v5.10.0

Hazelcast CR:
`apiVersion: hazelcast.com/v1alpha1
kind: Hazelcast
metadata:
  name: hazelcast-cluster
  namespace: xrox-hazelcast
spec:
  agent:
    repository: docker.io/hazelcast/platform-operator-agent
    version: 0.1.22
  clusterName: xrox
  clusterSize: 3
  exposeExternally:
    discoveryServiceType: LoadBalancer
    memberAccess: LoadBalancer
    type: Smart
  jvm:
    args:
    - -Dhazelcast.sql.experimental.custom.types.enabled=true
  loggingLevel: ALL
  managementCenter:
    consoleEnabled: true
    dataAccessEnabled: true
    scriptingEnabled: true
  repository: docker.io/hazelcast/hazelcast
  serialization:
    compactSerialization:
      serializers:
      - ar.com.arquant.hazelcast.trade.ExecutionReportMessageTSerializer
      - ar.com.arquant.hazelcast.pretrade.MarketDataSnapshotFullRefreshMessageTSerializer
  userCodeDeployment:
    clientEnabled: true
    configMaps:
    - execution-report-map-store
    remoteURLs:
    - http://hazelcast-bucket-rest-api-service.svmi.10.1.200.176.sslip.io/api/v1/download/fixml-repository.jar
    - http://hazelcast-bucket-rest-api-service.svmi.10.1.200.176.sslip.io/api/v1/download/fixml-hazelcast-repository.jar
    - http://hazelcast-bucket-rest-api-service.svmi.10.1.200.176.sslip.io/api/v1/download/jakarta.xml.bind.jaxb-api.jar
    - http://hazelcast-bucket-rest-api-service.svmi.10.1.200.176.sslip.io/api/v1/download/jakarta.xml.bind.jaxb-impl.jar
    - http://hazelcast-bucket-rest-api-service.svmi.10.1.200.176.sslip.io/api/v1/download/jakarta.xml.bind.jaxb-core.jar
    - http://hazelcast-bucket-rest-api-service.svmi.10.1.200.176.sslip.io/api/v1/download/jakarta.activation-api.jar
    - http://hazelcast-bucket-rest-api-service.svmi.10.1.200.176.sslip.io/api/v1/download/postgresql.jar
    triggerSequence: "001"
  version: 5.3.5-slim`

Management Center CR:
`apiVersion: hazelcast.com/v1alpha1
kind: ManagementCenter
metadata:
  name: management-center
  namespace: hazelcast
spec:
  hazelcastClusters:
  - address: hazelcast-cluster.xrox-hazelcast
    name: xrox
  persistence:
    enabled: true
    size: 10G
  repository: hazelcast/management-center
  version: 5.3.3`

I'm using vanilla Kubernetes in a VMWare VSphere enviroment.

Operator Log
Management Center Log:
[management-center-0.log](https://github.com/hazelcast/hazelcast-platform-operator/files/14296480/management-center-0.log)

Failing Hazelcast POD Log after change:
`########################################
# JAVA=/usr/bin/java
# JAVA_OPTS=--add-modules java.se --add-exports java.base/jdk.internal.ref=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.management/sun.management=ALL-UNNAMED --add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED -Dhazelcast.logging.type=log4j2 -Dlog4j.configurationFile=file:/opt/hazelcast/config/log4j2.properties -Dhazelcast.config=/opt/hazelcast/config/hazelcast-docker.xml -Djet.custom.lib.dir=/opt/hazelcast/custom-lib -Djava.net.preferIPv4Stack=true -XX:MaxRAMPercentage=80.0 -XX:MaxGCPauseMillis=5 -Dhazelcast.config=/data/hazelcast/hazelcast.yaml -XX:+UseContainerSupport -Dhazelcast.sql.experimental.custom.types.enabled=true
# CLASSPATH=/opt/hazelcast/*:/opt/hazelcast/userCode/bucket/*:/opt/hazelcast/userCode/urls/*:/opt/hazelcast/userCode/cm/execution-report-map-store/*:/opt/hazelcast/lib:/opt/hazelcast/lib/*:/opt/hazelcast/bin/user-lib:/opt/hazelcast/bin/user-lib/*
########################################
{"time":"2024-02-15T11:56:20,744", "logger": "com.hazelcast.internal.util.JavaVersion", "level": "DEBUG", "msg": "Detected runtime version: Java 11 "}
{"time":"2024-02-15T11:56:20,765", "logger": "com.hazelcast.internal.config.AbstractConfigLocator", "level": "INFO", "msg": "Loading configuration '/data/hazelcast/hazelcast.yaml' from System property 'hazelcast.config' "}
{"time":"2024-02-15T11:56:20,766", "logger": "com.hazelcast.internal.config.AbstractConfigLocator", "level": "INFO", "msg": "Using configuration file at /data/hazelcast/hazelcast.yaml "}
2 schema violations found
Usage: hz start [-hV] [-c=<file>] [-i=<interface>] [-p=<port>]
Starts a new Hazelcast member
  -h, --help            Show this help message and exit.
  -V, --version         Print version information and exit.
  -c, --config=<file>   Use <file> for Hazelcast configuration. Accepted
                          formats are XML and YAML.
  -p, --port=<port>     Bind to the specified <port>. Please note that if the
                          specified port is in use, it will auto-increment to
                          the first free port. (default: 5701)
  -i, --interface=<interface>
                        Bind to the specified <interface>.`

The main reason the pod is failing is because of this section of the generated yaml, I got this from the hazelcast secret.
`serialization:
        portable-version: 0
        use-native-byte-order: false
        enable-compression: false
        enable-shared-object: false
        allow-override-default-serializers: false
        allow-unsafe: false
        byte-order: BIG_ENDIAN
        compact-serialization:
            serializers:
                - 'serializer: ar.com.arquant.hazelcast.trade.ExecutionReportMessageTSerializer'
                - 'serializer: ar.com.arquant.hazelcast.pretrade.MarketDataSnapshotFullRefreshMessageTSerializer'
`
The single quotes are the problem.  It has to be a object not a string.

In other words it hast to be like this
`serializers:
    - serializer: ar.com.arquant.hazelcast.trade.ExecutionReportMessageTSerializer
    - serializer: ar.com.arquant.hazelcast.pretrade.MarketDataSnapshotFullRefreshMessageTSerializer`

**Comment by dalcorta** [comment]
The problem is in this file: https://github.com/hazelcast/hazelcast-platform-operator/blob/main/controllers/hazelcast/hazelcast_resources.go

The code problem is in this block:
```
if s.CompactSerialization != nil {
		classes := make([]string, 0, len(s.CompactSerialization.Classes))  <-- Should be an array of object not string
		for _, class := range s.CompactSerialization.Classes {
			classes = append(classes, fmt.Sprintf("class: %s", class))
		}
		serializers := make([]string, 0, len(s.CompactSerialization.Serializers))
		for _, serializer := range s.CompactSerialization.Serializers {
			serializers = append(serializers, fmt.Sprintf("serializer: %s", serializer))
		}
		cfg.Serialization.CompactSerialization = &config.CompactSerialization{
			Serializers: serializers,
			Classes:     classes,
		}
	}
```

**Comment by dalcorta** [comment]
![image](https://github.com/hazelcast/hazelcast-platform-operator/assets/2669695/85fa21b6-9274-4251-ab5e-404ad5f46395)
